programs/chat/src/instructions/send_message/send_token_message_v0.rs

```diff
@@ -10,13 +10,16 @@ use std::str::FromStr;
 #[derive(Accounts)]
 pub struct SendTokenMessageV0<'info> {
   pub chat: Box<Account<'info, ChatV0>>,
+  #[account(
+    has_one = chat
+  )]
+  pub chat_permissions: Box<Account<'info, ChatPermissionsV0>>,
```

Comment:
A fake chat_permissions account can be created and passed in here, which will allow unauthorised messages to be sent to a chat. Add seeds check.

Reply:
`has_one` prevents that. You can only create a chat permissions account if you are the admin of the chat

Reply:
Does anchor make sure a chat permissions account is created by the correct program ID by default?